aten/src/ATen/native/QuantizedLinear.cpp

```diff
@@ -32,6 +32,129 @@ namespace native {
 
 #ifdef USE_FBGEMM
 
+Tensor fbgemm_linear_int8_weight_fp32_activation(
+    const Tensor& input,
+    const Tensor& weight,
+    const Tensor& packed,
+    const Tensor& col_offsets,
+    Scalar weight_scale,
+    Scalar weight_zero_point,
+    const Tensor& bias) {
+  // We make a strong guarantee that models using these operators will have the
+  // same numerics across different machines. Therefore, we do not provide a
+  // fallback path and rather fail loudly if we cannot run FBGEMM.
+  TORCH_CHECK(fbgemm::fbgemmSupportedCPU(), "Your CPU doesn't support FBGEMM.");
+
+  auto input_contig = input.contiguous();
+  auto* input_ptr = input_contig.data<float>();
+
+  TORCH_CHECK(input.dim() >= 2);
+  int64_t M = size_to_dim_(input.dim() - 1, input.sizes());
```

Comment:
I mentioned it before -- single character variables are NOT ok! A general rule of thumb is, if it takes me more than a glance to understand what the variable is representing -- rename it. Every time I read this piece I have to go back and forth just to make sure there is no accidental typo bug.

Reply:
Follow your suggestion.

Reply:
K and N are the conventional names for dimensions in all GEMM routines: https://spatial-lang.org/gemm

We should revert back to using those identifiers, otherwise this will cause confusion

Reply:
On second thought, this section is pretty insane:

```
      /*nRow=*/batch_size,
      /*nCol=*/input_channels,
      /*smat=*/input_ptr,
      /*ld=*/input_channels,
```

Can we instead rename these variables to match the FBGEMM API (i.e. `nRow`, `nCol`, and `ld)?

Reply:
`nRow`, `nCol` needs to be connected to a specific matrix. For `fbgemm::PackAWithQuantRowOffset` (`PackA`), `nRow` is `M` or `batch_size`, and `nCol` is `K` or `input_channels`; For `fbgemm::PackBMatrix`, `nRow` is `K` or `input_channels` and `nCol` is `N` or `output_channels`. The name `ld` is for the leading dimension (https://www.youtube.com/watch?v=PhjildK5oO8), which also depends on whether the matrix is A, B, or C, as well as the matrix is stored in the row-major or column-major. It might cause confusions if we rename these variables as `nRow`, `nCol`, and `ld`

Reply:
@jamesr66a : I totally concur with you that we can just use `M`, `K`, and `N`, as they are the conventional names for dimensions in all GEMM routines (https://petewarden.com/2015/04/20/why-gemm-is-at-the-heart-of-deep-learning/). @zafartahirov : What do you think?

Reply:
The links to the blog posts are irrelevant in the case of the dev, because readability of the code is there to make the life of the devs easier, not the blog readers. The problem with names like `M`, `N`, `K` is that they force the readers to go back to the other parts of the code, just to clarify what was going on. The reason it is a standard within FBGEMM is not because it is good, but because it was not caught earlier in the development. 

However, it has a minimal impact on productivity. And although I am a strong proponent of descriptive variable names, I will submit to the better judgement of the FBGEMM devs (it doesn't mean I will not flag it in the future tho 😄 ).